framework/db/BaseActiveRecord.php

```diff
@@ -514,12 +514,13 @@ public function markAttributeDirty($name)
     /**
      * Returns a value indicating whether the named attribute has been changed.
      * @param string $name the name of the attribute
-     * @return boolean whether the attribute has been changed
+     * @param bool $identical if the comparison is made as identical
+     * @return bool whether the attribute has been changed
      */
-    public function isAttributeChanged($name)
+    public function isAttributeChanged($name, $identical = true)
     {
         if (isset($this->_attributes[$name], $this->_oldAttributes[$name])) {
-            return $this->_attributes[$name] !== $this->_oldAttributes[$name];
+            return $identical ? $this->_attributes[$name] !== $this->_oldAttributes[$name] : $this->_attributes[$name] != $this->_oldAttributes[$name];
```

Comment:
code looks confusing on one line, should be a normal if `if ($identical) ... else ...`